### PR TITLE
chore: minor camel-catalog cleanup

### PR DESCRIPTION
**Pull request by lburgazzoli**
# Description

# Target

- [ ] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [ ] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [ ] I checked that each commit in the pull request has a meaningful subject line and body.

- [ ] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

### CODE REVIEW

File: catalog/camel-catalog/src/main/java/org/apache/camel/catalog/DefaultCamelCatalog.java [hunk 1]
```diff
@@ -113,13 +131,32 @@ public void setRuntimeProvider(RuntimeProvider runtimeProvider) {
...
+        cache.remove(LIST_OTHERS_AS_JSON);
+
+        cache.remove(FIND_MODEL_NAMES);
+        cache.remove(FIND_MODEL_LABELS);
+        cache.remove(LIST_MODELS_AS_JSON);
+
+        cache.remove(FIND_TRANSFORMER_NAMES);
+        cache.remove(LIST_TRANSFORMERS_AS_JSON);
+
+        cache.remove(SUMMARY_AS_JSON);
```
Comment (lburgazzoli) [inline catalog/camel-catalog/src/main/java/org/apache/camel/catalog/DefaultCamelCatalog.java:159]: @davsclaus I don't know if this was intentional but saw a number of element of the cache not being removed when a new runtime provider is set so I removed them. Maybe cleaning up the whole cache and re-cache some of the element upfront would be better ?

**Review by oscerd — APPROVED** [review]

File: catalog/camel-catalog/src/main/java/org/apache/camel/catalog/DefaultCamelCatalog.java (hunk 1 above)
Comment (davsclaus) [reply catalog/camel-catalog/src/main/java/org/apache/camel/catalog/DefaultCamelCatalog.java:159]: Yeah that is safer, and its no problem - also for changing provider, you would just create the provider up front and not change it runtime wise.

Comment (davsclaus) [reply catalog/camel-catalog/src/main/java/org/apache/camel/catalog/DefaultCamelCatalog.java:159]: You are welcome to clear all cache, or create a JIRA so we wont forget

Comment (lburgazzoli) [reply catalog/camel-catalog/src/main/java/org/apache/camel/catalog/DefaultCamelCatalog.java]: done

**Pull request merged**